packages/blade/src/components/BladeProvider/BladeProvider.tsx

```diff
@@ -29,9 +34,12 @@ const BladeProvider = ({
   }
 
   const { colorScheme, setColorScheme } = useColorScheme(initialColorScheme);
+  const { matchedDeviceType } = useBreakpoint({
+    breakpoints: themeTokens.breakpoints,
+  });
```

Comment:
This registers an event listener for media query changes. So anytime the media query changes it'll re-render the entire children tree wrapped under `BladeProvider`